### DISCUSSION TITLE
Bootstrap Vcpkg prior to FetchContent

### DISCUSSION THREAD

**Discussion by cursey** [build-cpp/cmkr#42] [Ideas]
I have a project that uses Vcpkg for all of its dependencies. I want to add a new dependency to another project of mine via FetchContent. Currently, because cmkr does all of its `FetchContent_MakeAvailable` calls prior to bootstraping Vcpkg, my new dependency added via FetchContent breaks because it cannot find one of the libraries it depends on (which is available through Vcpkg).

This is fixed if bootstrapping Vcpkg happens prior to any `FetchContent_MakeAvailable` calls. I'm not super familiar with FetchContent and bootstrapping Vcpkg in general though so maybe I'm missing something here and there is a better way for me to do this.

Any thoughts on this? If bootstrapping Vcpkg prior to any FetchContent packages is the right way to go I would be happy to implement this and open a PR.

**Comment by mrexodia** [comment]
I guess switching the order would be fine, but I don’t think it’s a good idea to mix dependency strategies. Probably best to do everything through vcpkg or everything through FetchContent…

**Reply by cursey** [reply]
I understand. That's kind of what I figured. I'll look into consuming my library via vcpkg first then.

**Reply by cursey** [reply]
I'm going to be honest, consuming my library through Vcpkg was not an enjoyable experience. I had to make a local Vcpkg registry, setup port files, versioning files, baseline files, and create a `vcpkg-configuration.json` to use the registry. The entire process was super finicky and the Vcpkg documentation on doing this kind of thing was pretty lacking.

You can see the PR here: https://github.com/cursey/regenny/pull/4

All this just to consume a header only library that itself uses cmkr. I would prefer a simpler solution to this since cmkr is all about making things easy. I feel my original idea of just bootstrapping Vcpkg prior to FetchContent is a good compromise. I would like to hear your thoughts on this or if you have any experience dealing with this type of thing in a more elegant way.

**Reply by mrexodia** [reply]
Yeah that looks like a total pain in the ass… I don’t think there’s any downside to bootstrapping vcpkg first (since it’s just a FetchContent call in the end), so feel free to PR if you want :)

As a side note, perhaps part of #16 could be a `vcpkg = true` to automatically generate the relevant files for vcpkg because that seems to be a lot of work for nothing…

**Reply by cursey** [reply]
I've created a PR https://github.com/build-cpp/cmkr/pull/45. Here's the PR on my own project to compare using FetchContent for my own library vs using Vcpkg repositories:

Using Vcpkg repositories: https://github.com/cursey/regenny/pull/4
Using FetchContent: https://github.com/cursey/regenny/pull/5

I am much happier with FetchContent and cmkr in this case.
